infrastructures/openshift/src/main/java/org/eclipse/che/workspace/infrastructure/openshift/environment/OpenShiftEnvironmentFactory.java

```diff
@@ -111,6 +111,7 @@ protected OpenShiftEnvironment doCreate(
         throw new ValidationException("Supporting of deployment configs is not implemented yet.");
       } else if (object instanceof Pod) {
         Pod pod = (Pod) object;
+        checkNotNull(pod.getMetadata(), "Pod metadata must not be null");
```

Comment:
I think the same should be done for other objects. Also, I see no check that name inside of metadata is not null or empty while it must not be.
Maybe it would be better have somehow util method that will safely retrieve name of `HasMetadata` object?
I mean something like that
```
  public String getName(HasMetadata object) {
    ObjectMeta metadata = object.getMetadata();
    checkNotNull(metadata, String.format("%s metadata must not be null", object.getKind()));
    String name = metadata.getName();
    checkNotNull(metadata, String.format("%s name must not be null", object.getKind()));
    return name;
  }
```
Then we would be able to reuse it for all kinds of K8s/OpenShift objects. WDYT?